### PR TITLE
doc: Add documentation for CMake `protobuf_generate`

### PR DESCRIPTION
**Pull request by Tradias**
How to use it.
How it works. (basic example and an example with gRPC)
Reference documentation.

Resolves https://github.com/protocolbuffers/protobuf/issues/10303

### CODE REVIEW

File: docs/cmake_protobuf_generate.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+Then building `proto-objects` will generate and compile:
+
+- `${CMAKE_CURRENT_BINARY_DIR}/generated/helloworld/helloworld.pb.h`
+- `${CMAKE_CURRENT_BINARY_DIR}/generated/helloworld/helloworld.pb.cc`
+- `${CMAKE_CURRENT_BINARY_DIR}/generated/helloworld/helloworld.grpc.pb.h`
+- `${CMAKE_CURRENT_BINARY_DIR}/generated/helloworld/helloworld.grpc.pb.cc`
+
+And `protoc` will automatically be re-run whenever the schema files change and `proto-objects` is build.
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:88]: "...is built."

File: docs/cmake_protobuf_generate.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Flag arguments:
+
+- `APPEND_PATH` — A flag that causes the base path of all proto schema files to be added to `IMPORT_DIRS`.
+
+Single value arguments:
+
+- `LANGUAGE` — A single value: cpp or python. Determines what kind of source files are being generated.
+- `OUT_VAR` — Name of a CMake variable that will be filled with the paths to the generated source files.
+- `EXPORT_MACRO` — Name of a macro that is applied to all generated Protobuf message classes and extern variables. It can e.g. be used to declare DLL exports.
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:123]: "It can, for example, be used to..."

File: docs/cmake_protobuf_generate.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+- `PROTOC_OUT_DIR` — Output directory of generated source files, defaults to `CMAKE_CURRENT_BINARY_DIR`.
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:124]: "...source files. Defaults to..."

File: docs/cmake_protobuf_generate.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+- `PLUGIN` — An optional plugin executable. This could e.g. be the path to `grpc_cpp_plugin`.
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:125]: "...could, for example, be the path..."

File: docs/cmake_protobuf_generate.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+- `PLUGIN_OPTIONS` — Additional options provided to the plugin, e.g. `generate_mock_code=true` for the gRPC cpp plugin.
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:126]: "...plugin, such as..."

File: docs/cmake_protobuf_generate.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+- `DEPENDENCIES` — Arguments forwarded to the `DEPENDS` of the underlying `add_custom_command` invocation.
+- `TARGET` — CMake target that will have the generated files added as sources.
+
+Multi value arguments:
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:130]: "Multi-value"

File: docs/cmake_protobuf_generate.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+## Reference
+
+Arguments accepted by `protobuf_generate`.
+
+Flag arguments:
+
+- `APPEND_PATH` — A flag that causes the base path of all proto schema files to be added to `IMPORT_DIRS`.
+
+Single value arguments:
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:119]: "Single-value"

File: docs/cmake_protobuf_generate.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+- `PROTOC_OUT_DIR` — Output directory of generated source files, defaults to `CMAKE_CURRENT_BINARY_DIR`.
+- `PLUGIN` — An optional plugin executable. This could e.g. be the path to `grpc_cpp_plugin`.
+- `PLUGIN_OPTIONS` — Additional options provided to the plugin, e.g. `generate_mock_code=true` for the gRPC cpp plugin.
+- `DEPENDENCIES` — Arguments forwarded to the `DEPENDS` of the underlying `add_custom_command` invocation.
+- `TARGET` — CMake target that will have the generated files added as sources.
+
+Multi value arguments:
+
+- `PROTOS` — List of proto schema files. If omitted, then every source file ending in *proto* of `TARGET` will be used.
+- `IMPORT_DIRS` — A common parent directory for the schema files. E.g., if the schema file is `proto/helloworld/helloworld.proto` and the import directory `proto/` then the generated files are `${PROTOC_OUT_DIR}/helloworld/helloworld.pb.h` and `${PROTOC_OUT_DIR}/helloworld/helloworld.pb.cc`.
```
Comment (Logofile) [inline docs/cmake_protobuf_generate.md:133]: "For example, if the schema..."

**Review by Logofile — CHANGES_REQUESTED** [review]

**Comment by Tradias** [comment]
@Logofile I have addressed your requested changes. Thanks a lot for the review!

**Review by acozzette — APPROVED** [review]

**Review by Logofile — APPROVED** [review]

**Pull request merged**